core/io/resource_uid_db.cpp

```diff
@@ -0,0 +1,550 @@
+		}
+
+		Variant v = cfg.get_value(section, key);
+		if (v.get_type() != Variant::DICTIONARY) {
+			continue;
+		}
+		Dictionary d = v;
+
+		String stored_path = d.get("last_known_path", String());
+		String stored_md5 = d.get("md5", String());
+		int64_t stored_size = (int64_t)d.get("size", (int64_t)0);
+		String filename = d.get("filename", String());
+
+		bool entry_changed = false;
+		bool path_exists = false;
+		String final_path = stored_path;
+
+		if (!stored_path.is_empty() && FileAccess::exists(stored_path)) {
+			path_exists = true;
+			final_path = stored_path;
+		} else {
+			String recovered;
+			if (!filename.is_empty()) {
+				if (scan_for_filename(resource_root, filename, recovered)) {
+					final_path = recovered;
+					path_exists = true;
+				}
+			}
+			if (!path_exists && !stored_md5.is_empty() && stored_size > 0) {
+				String bycontent;
+				if (scan_for_md5_size(resource_root, stored_md5, stored_size, bycontent)) {
+					final_path = bycontent;
+					path_exists = true;
+				}
+			}
+		}
+
+		String cur_md5;
+		int64_t cur_size = 0;
+		if (!final_path.is_empty() && FileAccess::exists(final_path)) {
+			cur_md5 = FileAccess::get_md5(final_path);
+			Ref<FileAccess> fa = FileAccess::open(final_path, FileAccess::READ);
+			if (fa.is_valid()) {
+				cur_size = (int64_t)fa->get_length();
+			}
+		}
+
+		if (d.get("last_known_path", String()) != final_path) {
+			// append history only when path changed
+			Array hist = d.get("history", Array());
+
+			// avoid duplicate successive history entries
+			bool should_append = true;
+			if (hist.size() > 0) {
+				Variant last = hist[hist.size() - 1];
+				if (last.get_type() == Variant::DICTIONARY) {
+					Dictionary lastd = last;
+					String last_path = lastd.get("path", String());
+					if (last_path == final_path) {
+						should_append = false;
+					}
+				}
+			}
+
+			if (should_append) {
+				Dictionary h;
+				h["path"] = final_path;
+				h["datetime"] = now_iso;
+				hist.append(h);
+				d["history"] = hist;
+			}
+
+			d["last_known_path"] = final_path;
+			entry_changed = true;
+		}
+
+		if (d.get("md5", String()) != cur_md5) {
+			d["md5"] = cur_md5;
+			entry_changed = true;
+		}
+		if ((int64_t)d.get("size", (int64_t)0) != cur_size) {
+			d["size"] = cur_size;
+			entry_changed = true;
+		}
+
+		String filename_field = d.get("filename", String());
+		if (filename_field.is_empty() && !final_path.is_empty()) {
+			d["filename"] = final_path.get_file();
+			entry_changed = true;
+		}
+
+		d["last_seen_on_ts"] = now_ts;
+		d["last_seen_on"] = now_iso;
+
+		if (final_path.is_empty() || !FileAccess::exists(final_path)) {
+			d["missing"] = true;
+		} else {
+			if (d.has("missing")) {
+				d.erase("missing");
+			}
+		}
+
+		if (entry_changed) {
+			cfg.set_value(section, key, d);
+			entries.insert(key, d);
+			any_modified = true;
+		} else {
+			entries.insert(key, d);
+		}
+	}
+
+	if (any_modified) {
+		save_db();
+	}
+	print_line("UIDDB: Refresh complete.");
```

Comment:
```suggestion
	print_verbose("UIDDB: Refresh complete.");
```
If we keep this, too much noise otherwise